app/models/activity_insight_oa_file.rb

```diff
@@ -16,6 +16,7 @@ class ActivityInsightOAFile < ApplicationRecord
       .where(file_download_location: nil)
```

Comment:
I'm curious about this since I added it to my PR for the file version checking job.  Should we only be filtering out open access locations from ScholarSphere here now?

Reply:
If so, I have it in my PR and it's tested so we don't need to do it here.

Reply:
@ajkiessl Ah, yes - that's right. Good catch.

There's no point in downloading a file for a publication that we already know to be in ScholarSphere since those publications drop completely out of the workflow as soon as we have that information, but we do want to download files for publications that have _other_ kinds of open access locations _unless_ we know that the OA status of those publications is gold or hybrid (line 19).

I didn't really think about whether or not we want to exclude files for publications that don't have an open access status yet (also line 19 here). I need to give that a little more thought when I have a minute..

Reply:
@ajkiessl You're right. That is something that can just be resolved when it's merged right?

@EricDurante my thought process was we should exclude nulls because null means it hasn't been checked yet & could be gold/hybrid. If it has been checked & we couldn't find a status, it would be `unknown`.

Reply:
Yeah, my stuff should be merged soon, so as long as they're both merged before the next release we are good.

Reply:
@Smullz622 Yeah, your reasoning totally makes sense. The thing that I was mulling over was whether or not we should go ahead and download the file anyway - even if the OA status hasn't been checked yet.

We're currently not considering whether or not we've checked for ScholarSphere OA locations (which is probably fine), we're only considering whether or not we currently know about any. We could likewise forget about considering whether or not we've checked for OA status yet and only consider whether or not it's currently gold or hybrid. Or we could go the other direction and make the file download wait until we know that we've checked ScholarSphere too.

I guess a downside of forcing the workflow to be more linear by waiting on the other checks to do the file download is that less things can potentially happen in parallel, and the whole process might take a little more time for a given publication.

A downside of allowing the file download (and dependent version checking, permissions checking, etc.) to happen in parallel with the steps for DOI checking, OA metadata checking, etc. is that we might waste a little computing power on doing those steps for publications that might get booted from the workflow later.

Another (and bigger) downside of making the file download wait for the OA metadata check is that admins won't yet have the file available to look at if the publication ends up in the manual DOI check list. DOI verification isn't currently a prerequisite for the file downloading, but it is for the OA metadata checking. So if we make OA metadata checking a prerequisite for the file download, then DOI becomes prerequisite too.  I think that having the file sometimes helps with that manual DOI verification.

Having thought this through, I'm pretty sure that we want to go ahead and download the file even if the OA status of the publication is still NULL.

Do you all agree?